sql/executor.go

```diff
@@ -165,28 +165,47 @@ func (e Executor) execStmt(stmt parser.Statement, params parameters, planMaker *
 				} else {
 					switch vt := val.(type) {
 					case parser.DBool:
-						row.Values = append(row.Values, driver.Datum{BoolVal: (*bool)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_BoolVal{BoolVal: bool(vt)},
```

Comment:
We can batch the allocations of the various `driver.Datum_*Val` structs. Something like:

```
  var boolVals []driver.Datum_BoolVal
  ...
  switch ... {
  case parser.DBool:
    if len(boolVals) == 0 {
      boolVals = make([]driver.Datum_BoolVal, 64)
    }
    boolVal := &boolVals[0]   
    boolVals = boolVals[1:]
    boolVal.BoolVal = bool(vt)
    row.Values = append(row.Values, driver.Datum{Payload: boolVal})
  }
```

A bit irritating to code this way, but if possible if we determine these allocations are a significant overhead.